<?xml version="1.0"?>
<launch>
    <param name="use_sim_time" value="true"/>

    <!--Nodes-->
    <node name="ipg_mediator" pkg="fs_planning-2022" type="ipg_mediator.py" output="screen"/>
	<node name="path_planner" pkg="fs_planning-2022" type="path_planner.py" output="screen"/>

    <!--Topics-->
    <param name="planner/map_topic" value="/planner/cones"/> 
    <param name="planner/pose_topic" value="/planner/pose"/>
    <param name="planner/waypoints_topic" value="/planner/path"/> 

    <!--Cones Styles-->
    <param name="planner/blue_cone_style" value="0"/>
    <param name="planner/yellow_cone_style" value="1"/>
    <param name="planner/orange_cone_style" value="2"/>
    <param name="planner/big_cone_style" value="3"/>
    <param name="planner/unkown_cone_style" value="4"/>
    
    <!--Mediator Parameters-->
    <param name="planner/add_noise" value="false"/>
    <param name="planner/mediator_map_topic" value="/ObjectList"/> <!--Topic to which mediator node subscribes for cones-->
    <param name="planner/mediator_pose_topic" value="/IMU"/> <!--Topic to which mediator node subscribes for pose-->

    <!--Cost Parameters-->
    <param name="planner/prior_weight" value="1"/>
    <param name="planner/largest_angle_change_weight" value="1"/>
    <param name="planner/largest_angle_change_setpoint" value="0"/>
    <param name="planner/left_standard_deviation_weight" value="1"/> 
    <param name="planner/left_standard_deviation_setpoint" value="0"/> 
    <param name="planner/right_standard_deviation_weight" value="1"/> 
    <param name="planner/right_standard_deviation_setpoint" value="0"/> 
    <param name="planner/track_width_standard_deviation_weight" value="1"/> 
    <param name="planner/track_width_standard_deviation_setpoint" value="0"/> 
    <param name="planner/number_of_edges_crossed_weight" value="100"/> 
    <param name="planner/planner/number_of_edges_crossed_setpoint" value="5"/> 
    <param name="planner/path_length_weight" value="20"/> 
    <param name="planner/path_length_setpoint" value="20"/> 

    <!--Path Finder Parameters-->
    <param name="planner/cone_field_of_view" value="180"/>
    <param name="planner/cone_distance" value="20"/>
    <param name="planner/waypoint_field_of_view" value="180"/>
    <param name="planner/waypoint_distance" value="6"/> 
    <param name="planner/path_queue_limit" value="10"/> 
    <param name="planner/max_waypoints_per_path" value="10"/> 

</launch>

